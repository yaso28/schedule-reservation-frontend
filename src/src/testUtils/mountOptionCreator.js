import { createLocalVue } from '@vue/test-utils'
import VueRouter from 'vue-router'
import routes from '@/router/routes'
import Vuex from 'vuex'
import VueI18n from 'vue-i18n'
import i18nOptions from '@/i18n/options'
import VueMeta from 'vue-meta'
import CoreuiVue from '@coreui/vue'
import icons from '@/icons'
import services from '@/services'

class MountOptionCreator {
  create (storeModules = null) {
    const localVue = createLocalVue()

    // router
    localVue.use(VueRouter)
    const router = new VueRouter({
      mode: 'history',
      base: process.env.BASE_URL,
      routes
    })

    // i18n
    localVue.use(VueI18n)
    const i18n = new VueI18n(i18nOptions)

    // vue-meta
    localVue.use(VueMeta)

    // Core UI
    localVue.use(CoreuiVue)

    const option = {
      localVue,
      router,
      i18n,
      icons,
      provide: services
    }

    // store
    if (storeModules) {
      localVue.use(Vuex)
      option.store = new Vuex.Store({
        modules: storeModules
      })
    }

    return option
  }
}

export default new MountOptionCreator()
