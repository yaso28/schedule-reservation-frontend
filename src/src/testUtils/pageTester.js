import { mount } from '@vue/test-utils'
import MyPage from '@/components/containers/MyPage'

class PageTester {
  async mount (pageComponent, route, mountOption) {
    await mountOption.router.push(route)
    const pageWrapper = mount(MyPage, mountOption)
    return pageWrapper.findComponent(pageComponent)
  }

  assertTitleTag () {
    expect(document.title).toMatchSnapshot()
  }

  assertMetaTags () {
    expect(document.querySelectorAll('meta[data-vue-meta]')).toMatchSnapshot()
  }
}

export default new PageTester()
