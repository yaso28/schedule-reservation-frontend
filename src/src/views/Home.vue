<template>
  <div class="home">
    <h1>{{ $t('organization_name') }} {{ $t('app_name') }}</h1>

    <CCard
      accent-color="primary"
      class="mt-4"
    >
      <CCardBody>
        <p>
          練習会場予約状況の管理および練習予定の公開を行うWebシステムです。<br>
          （ポートフォリオ用のデモサイトです。）
        </p>

        <div class="my-2">
          <CButton
            color="primary"
            variant="outline"
            @click="showGuest = !showGuest"
          >
            ゲストができること
          </CButton>
          <CCollapse :show="showGuest">
            <CListGroup>
              <CListGroupItem
                v-for="item in guestActions"
                :key="item.text"
                class="text-body"
                :to="item.to"
              >
                {{ item.text }}
              </CListGroupItem>
            </CListGroup>
          </CCollapse>
        </div>

        <div class="my-2">
          <CButton
            color="primary"
            variant="outline"
            @click="showAdmin = !showAdmin"
          >
            システム管理者(*1)ができること
          </CButton>
          <CCollapse :show="showAdmin">
            <CListGroup>
              <CListGroupItem
                v-for="item in adminActions"
                :key="item.text"
                class="text-body"
                :to="item.to"
              >
                {{ item.text }}
              </CListGroupItem>
            </CListGroup>
            <ul class="list-unstyled ml-1">
              <li>
                (*1) ログインID：admin パスワード：admin$28
              </li>
              <li>
                (*2) 先行予約できるのは合計＊時間までといった上限がある施設にて時間帯を分けて予約する場合に利用します。<br>
                （例：6時間のうち3時間を先行予約しておき、1ヶ月前を過ぎ上限が無くなってから残り3時間を予約する。）
              </li>
            </ul>
          </CCollapse>
        </div>

        <div class="my-2">
          <CButton
            color="info"
            variant="outline"
            @click="showSource = !showSource"
          >
            ソースコード
          </CButton>
          <CCollapse :show="showSource">
            <CListGroup>
              <CListGroupItem
                v-for="item in sourceLinks"
                :key="item.text"
                class="text-body"
                :href="item.href"
                target="_blank"
              >
                {{ item.text }}
              </CListGroupItem>
            </CListGroup>
          </CCollapse>
        </div>
      </CCardBody>
    </CCard>
  </div>
</template>

<script>
export default {
  name: 'MyHome',
  data () {
    return {
      showGuest: false,
      showAdmin: false,
      showSource: false
    }
  },
  computed: {
    guestActions () {
      return [
        ['練習予定の閲覧', 'schedule.list']
      ].map(item => ({ text: item[0], to: { name: item[1] } }))
    },
    adminActions () {
      return [
        ['練習予定の追加・編集', 'reservation.schedule.list'],
        ['会場予約の編集・分割(*2)', 'reservation.reservation.list'],
        ['月別練習予定のメール送信', 'reservation.month.list']
      ].map(item => ({ text: item[0], to: { name: item[1] } }))
    },
    sourceLinks () {
      return [
        ['フロントエンド', 'https://github.com/yaso28/schedule-reservation-frontend'],
        ['バックエンド', 'https://github.com/yaso28/schedule-reservation-backend']
      ].map(item => ({ text: item[0], href: item[1] }))
    }
  }
}
</script>
