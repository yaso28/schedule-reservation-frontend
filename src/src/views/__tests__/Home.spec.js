import Home from '../Home'
import mountOptionCreator from '@/testUtils/mountOptionCreator'
import pageTester from '@/testUtils/pageTester'
import storeModules from '@/store/modules'

jest.mock('@/store/modules')

describe('Home', () => {
  let mountOption
  beforeEach(() => {
    mountOption = mountOptionCreator.create(storeModules)
  })

  describe('マウント', () => {
    let wrapper
    beforeEach(async () => {
      wrapper = await pageTester.mount(Home, { name: 'home' }, mountOption)
    })

    test('title', () => {
      pageTester.assertTitleTag()
    })

    test('meta tags', () => {
      pageTester.assertMetaTags()
    })

    test('ページ全体表示', () => {
      expect(wrapper.element).toMatchSnapshot()
    })
  })
})
