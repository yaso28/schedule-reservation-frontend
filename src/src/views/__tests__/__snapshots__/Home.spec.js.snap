// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home マウント meta tags 1`] = `
NodeList [
  <meta
    content="やそスポーツクラブの公式サイトです。"
    data-vmid="description"
    data-vue-meta="2"
    name="description"
  />,
]
`;

exports[`Home マウント title 1`] = `"やそスポーツクラブ | 公式サイト"`;

exports[`Home マウント ページ全体表示 1`] = `
<div
  class="home"
>
  <h1>
    やそスポーツクラブ 公式サイト
  </h1>
   
  <div
    class="card mt-4 card-accent-primary"
  >
    <div
      class="card-body"
    >
      <p>
        
        練習会場予約状況の管理および練習予定の公開を行うWebシステムです。
        <br />
        
        （ポートフォリオ用のデモサイトです。）
      
      </p>
       
      <div
        class="my-2"
      >
        <button
          class="btn btn-outline-primary"
          type="button"
        >
          
          ゲストができること
        
        </button>
         
        <div
          class=""
          style="display: none;"
        >
          <ul
            class="list-group"
            role="list-items"
          >
            <a
              class="list-group-item text-body list-group-item-action"
              href="/schedule/list"
              target="_self"
            >
              
              練習予定の閲覧
            
            </a>
          </ul>
        </div>
      </div>
       
      <div
        class="my-2"
      >
        <button
          class="btn btn-outline-primary"
          type="button"
        >
          
          システム管理者(*1)ができること
        
        </button>
         
        <div
          class=""
          style="display: none;"
        >
          <ul
            class="list-group"
            role="list-items"
          >
            <a
              class="list-group-item text-body list-group-item-action"
              href="/reservation/schedule/list"
              target="_self"
            >
              
              練習予定の追加・編集
            
            </a>
            <a
              class="list-group-item text-body list-group-item-action"
              href="/reservation/reservation/list"
              target="_self"
            >
              
              会場予約の編集・分割(*2)
            
            </a>
            <a
              class="list-group-item text-body list-group-item-action"
              href="/reservation/month/list"
              target="_self"
            >
              
              月別練習予定のメール送信
            
            </a>
          </ul>
           
          <ul
            class="list-unstyled ml-1"
          >
            <li>
              
              (*1) ログインID：admin パスワード：admin$28
            
            </li>
             
            <li>
              
              (*2) 先行予約できるのは合計＊時間までといった上限がある施設にて時間帯を分けて予約する場合に利用します。
              <br />
              
              （例：6時間のうち3時間を先行予約しておき、1ヶ月前を過ぎ上限が無くなってから残り3時間を予約する。）
            
            </li>
          </ul>
        </div>
      </div>
       
      <div
        class="my-2"
      >
        <button
          class="btn btn-outline-info"
          type="button"
        >
          
          ソースコード
        
        </button>
         
        <div
          class=""
          style="display: none;"
        >
          <ul
            class="list-group"
            role="list-items"
          >
            <a
              class="list-group-item text-body list-group-item-action"
              href="https://github.com/yaso28/schedule-reservation-frontend"
              rel="noopener"
              target="_blank"
            >
              
              フロントエンド
            
            </a>
            <a
              class="list-group-item text-body list-group-item-action"
              href="https://github.com/yaso28/schedule-reservation-backend"
              rel="noopener"
              target="_blank"
            >
              
              バックエンド
            
            </a>
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;
